microraiden/microraiden/client/http_client.py

```diff
@@ -83,65 +84,84 @@ def _request_resource(self, requested_resource: str) -> Tuple[Any, bool]:
             return None, False  # user requested abort
 
         if response.status_code == requests.codes.OK:
-            retry = self.on_success(response.content, headers.get('cost'))
-            return response.content, retry
+            return response.content, self.on_success(response.content, headers.get('cost'))
+
         elif response.status_code == requests.codes.PAYMENT_REQUIRED:
             if 'insuf_confs' in headers:
-                retry = self.on_insufficient_confirmations()
+                return None, self.on_insufficient_confirmations()
+
             elif 'insuf_funds' in headers:
-                retry = self.on_insufficient_funds()
+                return None, self.on_insufficient_funds()
+
+            elif 'contract_address' not in headers or not is_same_address(
+                headers.contract_address,
+                self.client.channel_manager_address
+            ):
+                return None, self.on_invalid_contract_address(
+                    headers.contract_address
+                )
+
             elif 'invalid_amount' in headers:
-                retry = self.on_invalid_amount()
-                self.channel.balance = int(headers.sender_balance)
-            else:
-                balance = None
-                balance_sig = None
+                last_balance = None
                 if 'sender_balance' in headers:
-                    balance = int(headers.sender_balance)
+                    last_balance = int(headers.sender_balance)
+                balance_sig = None
                 if 'balance_signature' in headers:
                     balance_sig = decode_hex(headers.balance_signature)
-                retry = self.on_payment_requested(
+                return None, self.on_invalid_amount(int(headers.price), last_balance, balance_sig)
+
+            else:
+                return None, self.on_payment_requested(
                     headers.receiver_address,
                     int(headers.price),
-                    balance,
-                    balance_sig,
-                    headers.get('contract_address')
                 )
-            return None, retry
         else:
-            return None, True
+            return None, self.on_http_error(requested_resource, response)
 
     def on_init(self, requested_resource):
-        pass
+        log.info('Starting request loop for resource at {}.'.format(requested_resource))
 
     def on_exit(self):
         pass
 
-    def on_success(self, resource, cost: int):
+    def on_success(self, resource, cost: int) -> bool:
+        log.info('Resource received.')
```

Comment:
Infos like these should be changed to debug or removed completely.

http://www.catb.org/esr/writings/taoup/html/ch01s06.html#id2878450